CityOfDetroit/bloom
# 1311/adding utility options

# Pull Request Template

## Issue Overview

This PR addresses #1311. 

This PR adds the option (not required) to indicate which utilities are included in the cost of rent. This is seen on the partners form fields, the listing detailed view on the partners side, and the detailed listing view on the public side. The public side representation in the AdditionalFees component required a prop refactoring so that a user can pass in an array of ReactNodes or strings to be displayed in a 2 column-format. Also, I expanded a few info-card__column css classes so that all sections of AdditionalFees.tsx resize in a consistent way and so that the classes could be potentially used for dynamic column representations in components other than AdditionalFees.tsx.

On the backend, this PR required a migration to add the new 'utilities' table to the model along with multiple updates to various backend type files. Additionally, translations and tests were added to complete and verify these changes.


## How Can This Be Tested/Reviewed?
Testing on Partners:
Part 1: This can be tested by selecting a listing, clicking edit, and scrolling down to ensure that the new utilities form fields are available and clickable. After selecting a variety of utilities, click 'save & exit', and then ensure that they are on the detailed view. 

Part 2: Go back to edit the same listing, unselect all options for utilities, and then save the listing again. The 'Utilities Included' title and bullets should be no longer visible.

Testing on Public:
Select a listing that you recently marked as having included utilities and observe the bulleted list in the 'Additional Fees' Component. If other 'additional fee' messages are available for the listing, the section will be in a 2-column format (with the exception of the last row which will fill if there is an odd number of footer info pieces).


## Checklist:


## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.
